### DISCUSSION TITLE
how to use node library with pb_hooks?

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#4844] [Q&A] [answered]
Hi guys, I'm trying to make this library work with pocketbase, but I'm not having success, can anyone help me?

```js
const nameToImdb = require(`${__hooks}/node_modules/name-to-imdb`);
```

I'm doing it this way:
```js
routerAdd("GET", "/refresh/movies", async (c) => {
  const { parseM3U } = require(`${__hooks}/node_modules/@iptv/playlist`)
  const nameToImdb = require(`${__hooks}/node_modules/name-to-imdb`);

  const url = 'my_api_url';

  try {
    console.log('Starting download...');

    const response = $http.send({
      url,
      method: 'GET',
      headers: {
      "content-type": "text/html",
      }
    })

    const rawData = response.raw;

    console.log('Download completed!');

    const playlist = parseM3U(rawData);

    const movies = playlist.channels.filter(channel => channel.groupTitle.includes('Filmes |'));

    const moviesWithImdb = await Promise.all(movies.map(movie =>
      new Promise((resolve, reject) => {
        nameToImdb(movie.name, (err, res) => {
          if (err) reject(err);
          resolve({ ...movie, imdbId: res });
        });
      })
    ));

    console.log('IMDb IDs fetched!');

    console.log('Movies with IMDb:', moviesWithImdb);

    return c.json(200, {
      message: 'Download completed!',
      length: moviesWithImdb.length,
    });

  } catch (error) {
    console.error('Error while downloading M3U content:', error);
    return c.json(500, { message: 'Error while downloading M3U content' });
  }
})
```

**Comment by ganigeorgiev** [comment]
`name-to-imdb` will not work because its dependencies rely on node-specific APIs.

Only pure JS ES5 modules are supported by the JSVM.

**Reply by bberenberg** [reply]
Is there a good way to check if a module will be supported by the JSVM? I found https://stackoverflow.com/questions/10538396/node-js-npm-anyway-to-tell-if-a-package-is-pure-js-or-not but not sure if theres a better approach I should be using?

Also is

```const nameToImdb = require(`${__hooks}/node_modules/name-to-imdb`);```

The correct way to reference modules if they are pure ES5?

**Reply by ganigeorgiev** [reply]
@bberenberg 

> Is there a good way to check if a module will be supported by the JSVM? 

I don't think there is an easy and reliable way to know upfront without checking against a list of node/browser only methods (and even then it may be difficult due to globals override, polyfills, etc.). In general, if the module needs to send network requests or access the file system then most likely is not a pure JS module because this functionality depends on the runtime.

> The correct way to reference modules if they are pure ES5?

The JSVM more-or-less follows the same rules as Node.js when loading modules with `require`. If you want to reference directly the module file, then providing a path to it should work. If you have a `node_modules` dir, using the name of the module should be sufficient (see also ["Loading modules"](https://pocketbase.io/docs/js-overview/#loading-modules)).

**Comment by brunowilliang** [comment]
@ganigeorgiev After this refactoring will it work? Or is that not in the plan?

**Reply by ganigeorgiev** [reply, marked as answer]
No sorry, there are no plans for this.

For more control over the application flow you'll have to switch to Go (or migrate to another platform - Supabase, Appwrite, Nhost, Directus, Strapi, PayloadCMS, EdgeDB, SurrealDB, etc.).
